### PR TITLE
fix: prevent UAF crash in setCertificateVerifyProc

### LINKED ISSUE #31675: [Bug]: Network service crashed, restarting service.
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

15.0.2

### What operating system are you using?

Ubuntu

### Operating System Version

Linux rrouwprlc0068 5.4.0-89-generic #100-Ubuntu SMP Fri Sep 24 14:50:10 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux, Ubuntu 20.04.3 LTS, GNOME Version 3.36.8, OS Type 64-bit

### What arch are you using?

x64

### Last Known Working Electron version

_No response_

### Expected Behavior

Network service should not crash and the error `[15:1102/154337.143944:ERROR:network_service_instance_impl.cc(333)] Network service crashed, restarting service.`, to not appear in the logs.

### Actual Behavior

Sporadically, when I run Electron inside a Docker container, I get the following error:

    [15:1102/154337.143944:ERROR:network_service_instance_impl.cc(333)] Network service crashed, restarting service.

Which causes partially rendering of the front-end application that is loaded with `win.loadURL` on localhost.

The issue seems very similar to https://bbs.archlinux.org/viewtopic.php?id=268123, but nothing from the suggestions works consinstently.

Also, when closing the Electron window, I get an error

    Error: ERR_FAILED (-2) loading 'https://localhost:{port}/{path_to}/index.html'
        at rejectAndCleanup (node:electron/js2c/browser_init:165:7486)
        at Object.stopLoadingListener (node:electron/js2c/browser_init:165:7861)
        at Object.emit (node:events:394:28) {
      errno: -2,
      code: 'ERR_FAILED',
      url: 'https://localhost:{port}/{path_to}/index.html'
    } [EOL]

Which is caught inside the main.js when executing await `win.loadURL(url)`

Also, in the browser window console, there is an error:

    ERROR Error: Uncaught (in promise): ChunkLoadError: Loading chunk 108 failed.
        (timeout: https://localhost:{port}/{path_to}/108.da4993913b4a27951976.js)
        ChunkLoadError: Loading chunk 108 failed.

Any ideas on the issue?

### Testcase Gist URL

_No response_

### Additional Information

Chromedriver 95.0.0 v

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
This fixes a bug with the RemoteCertVerifier patch that drives the
`ses.setCertificateVerifyProc` API. When an in-flight request was cancelled
after the procedure was invoked but before the result was received, a
use-after-free could occur.

Fixes #31675.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed a network service crash that could occur when using setCertificateVerifyProc.

### CODE REVIEW

File: patches/chromium/network_service_allow_remote_certificate_verification_logic.patch [hunk 1]
```diff
@@ -44,56 +70,42 @@ index 8ff62f92ed6efdbfc18db53db3c5bb59c1acfe34..ca62a13420aa9c114c00054bbe1215f9
 +  int Verify(const RequestParams& params,
 +             net::CertVerifyResult* verify_result,
 +             net::CompletionOnceCallback callback,
-+             std::unique_ptr<Request>* out_req,
++             std::unique_ptr<CertVerifier::Request>* out_req,
 +             const net::NetLogWithSource& net_log) override {
 +    out_req->reset();
++    *out_req = std::make_unique<Request>(); // TODO: also pass this to OnRequestFinished
```
Comment (deepak1556) [inline patches/chromium/network_service_allow_remote_certificate_verification_logic.patch:76]: Is the TODO comment still relevant ?

**Review by deepak1556 — APPROVED** [review]

File: patches/chromium/network_service_allow_remote_certificate_verification_logic.patch (hunk 1 above)
Comment (nornagon) [reply patches/chromium/network_service_allow_remote_certificate_verification_logic.patch:76]: nope, thanks for catching.

**Review by jkleinsc — CHANGES_REQUESTED** [review]
Looks like tests are legitimately failing.

**Review by deepak1556 — APPROVED** [review]

**Comment by lauw70** [comment]
@nornagon do you have any idea when we could expect this fix to land in a 17+.x.y stable release?

**Pull request merged**